Hi @adilburaksen,
  
  Thank you for your pull request and contribution to hardening `ImagePreprocessingStage`! The bounds validation and overflow checks are a great improvement.
  
  While importing this PR into Google's internal codebase, we ran into a C++ compile error in the test suite due to overload resolution ambiguity:
  
  
  image_preprocessing_stage_test.cc:308:11: error: call to member function 'AddCroppingStep' is ambiguous builder.AddCroppingStep(/width=/100000, /height=/100000);

  
  ### Explanation
  There are two overloaded signatures for `AddCroppingStep` in `image_preprocessing_stage.h`:
  1. `void AddCroppingStep(float cropping_fraction, bool use_square_cropping = false)`
  2. `void AddCroppingStep(uint32_t width, uint32_t height, bool use_square_cropping = false)`
  
  Passing signed integer literals (`100000`, `100000`) matches both overloads through standard implicit conversions, which makes the call ambiguous.
  
  ### Fix
  Could you please update the test call in `image_preprocessing_stage_test.cc` to use unsigned integer literals (`U` suffix)? E.g.:
  
  ```cpp
  builder.AddCroppingStep(/*width=*/100000U, /*height=*/100000U);
  
  Once you push this update to the branch, we will re-import the PR and merge it. Thank you again!

--- a/tensorflow/lite/tools/evaluation/stages/image_preprocessing_stage_test.cc
+++ b/tensorflow/lite/tools/evaluation/stages/image_preprocessing_stage_test.cc
@@ -305,7 +305,7 @@ TEST(ImagePreprocessingStage, TestCropTargetLargerThanImage) {
       std::string(kImagePreprocessingStageName), kTfLiteFloat32);
   // A crop target larger than the image would produce negative start offsets
   // and read out of bounds. It must be rejected at run time.
-  builder.AddCroppingStep(/*width=*/100000, /*height=*/100000);
+  builder.AddCroppingStep(/*width=*/100000U, /*height=*/100000U);
   builder.AddNormalizationStep(127.5, 1.0 / 127.5);
   ImagePreprocessingStage stage(builder.build());
   EXPECT_EQ(stage.Init(), kTfLiteOk);

